fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateRoutineLoadInfo.java

```diff
@@ -0,0 +1,525 @@
+            }
+        } else if (mergeType == LoadTask.MergeType.MERGE) {
+            throw new AnalysisException("Excepted DELETE ON clause when merge type is MERGE.");
+        }
+    }
+
+    private void checkDBTable(ConnectContext ctx) throws AnalysisException {
+        dbName = labelName.getDbName();
+        if (Strings.isNullOrEmpty(dbName)) {
+            dbName = ctx.getDatabase();
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+        }
+        name = labelName.getLabelName();
+        FeNameFormat.checkLabel(name);
+
+        Database db = Env.getCurrentInternalCatalog().getDbOrAnalysisException(dbName);
+        if (isPartialUpdate && isMultiTable) {
+            throw new AnalysisException("Partial update is not supported in multi-table load.");
+        }
+        if (isMultiTable) {
+            return;
+        }
+        if (Strings.isNullOrEmpty(tableName)) {
+            throw new AnalysisException("Table name should not be null");
+        }
+        Table table = db.getTableOrAnalysisException(tableName);
+        if (mergeType != LoadTask.MergeType.APPEND
+                && (table.getType() != Table.TableType.OLAP
+                || ((OlapTable) table).getKeysType() != KeysType.UNIQUE_KEYS)) {
+            throw new AnalysisException("load by MERGE or DELETE is only supported in unique tables.");
+        }
+        if (mergeType != LoadTask.MergeType.APPEND
+                && !(table.getType() == Table.TableType.OLAP && ((OlapTable) table).hasDeleteSign())) {
+            throw new AnalysisException("load by MERGE or DELETE need to upgrade table to support batch delete.");
+        }
+        if (isPartialUpdate && !((OlapTable) table).getEnableUniqueKeyMergeOnWrite()) {
+            throw new AnalysisException("load by PARTIAL_COLUMNS is only supported in unique table MoW");
+        }
+    }
+
+    private void checkLoadProperties(ConnectContext ctx) throws UserException {
+        Separator columnSeparator = null;
+        // TODO(yangzhengguo01): add line delimiter to properties
+        Separator lineDelimiter = null;
+        ImportColumnsStmt importColumnsStmt = null;
+        ImportWhereStmt precedingImportWhereStmt = null;
+        ImportWhereStmt importWhereStmt = null;
+        ImportSequenceStmt importSequenceStmt = null;
+        PartitionNames partitionNames = null;
+        ImportDeleteOnStmt importDeleteOnStmt = null;
+        CascadesContext cascadesContext = null;
+        ExpressionAnalyzer analyzer = null;
+        PlanTranslatorContext context = null;
+        if (!isMultiTable) {
+            List<String> nameParts = Lists.newArrayList();
+            nameParts.add(dbName);
+            nameParts.add(tableName);
+            Plan unboundRelation = new UnboundRelation(StatementScopeIdGenerator.newRelationId(), nameParts);
```

Comment:
bind an Expression and translate to legacy Expr is used very often. Better to put this into a new ExpressionToExprUtil class

Reply:
the initialize of translation differs a lot, so it is hard to extract one common analyzer and translator. In next pr using it I would extract common codes